src/test/java/org/jabref/gui/externalfiles/UnlinkedFilesCrawlerTest.java

```diff
@@ -35,4 +39,31 @@ public void minimalGitIgnore(@TempDir Path testRoot) throws Exception {
 
         assertEquals(new FileNodeViewModel(testRoot), fileNodeViewModel);
     }
+
+    @Test
+    public void excludingTheCurrentLibraryTest(@TempDir Path testRoot) throws IOException {
+        // Adding 3 files one of which is the database file
+        Files.createFile(testRoot.resolve("unlinkedPdf.pdf"));
+        Files.createFile(testRoot.resolve("another-unlinkedPdf.pdf"));
+        Path databasePath = testRoot.resolve("test.bib");
+        Files.createFile(databasePath);
+
+        BibDatabaseContext databaseContext = new BibDatabaseContext();
+        databaseContext.setDatabasePath(databasePath);
+
+        FilePreferences filePreferences = mock(FilePreferences.class);
+        Filter<Path> fileExtensionFilter = new FileExtensionViewModel(StandardFileType.ANY_FILE, filePreferences).dirFilter();
+        UnlinkedPDFFileFilter unlinkedPdfFileFilter = new UnlinkedPDFFileFilter(fileExtensionFilter, databaseContext, filePreferences);
+
+        UnlinkedFilesCrawler unlinkedFilesCrawler = new UnlinkedFilesCrawler(testRoot, unlinkedPdfFileFilter, DateRange.ALL_TIME, ExternalFileSorter.DEFAULT, databaseContext, filePreferences);
+        FileNodeViewModel fileNodeViewModel = unlinkedFilesCrawler.searchDirectory(testRoot, unlinkedPdfFileFilter);
+
+        // checking to see if the database file has been filtered
+        try (Stream<Path> filesInitially = Files.list(testRoot)) {
+            int count = (int) filesInitially.count();
+            assertEquals(fileNodeViewModel.getFileCount(), count - 1);
+        } catch (Exception e) {
```

Comment:
You can use a try with resources statement also without any exception handling, it's  then sufficient to add `throws Exception` to the test https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html